### PR TITLE
Adds __env__ substitution to ext_pillar.stack 

### PR DESCRIPTION
**Pull request by morgana2313**
### What does this PR do?

* Adds `__env__`substitution to ext_pillar.stack; followup of #61531 
* Improved exception handling for stacked template (jinja) template rendering and yaml parsing in ext_pillar.stack

### New Behavior

adds `__env__` to stacked pillar config

```
    ext_pillar:
      - stack:
          opts:pillarenv:
            dev: /path/to/dev/stack.cfg
            __env__: /path/to/__env__/stack.cfg
```

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

@nicholasmhughes: I think this code is working ok, do you agree?

### CODE REVIEW

**Comment by whytewolf** [comment]
can you rebase your branch against master from the start? it looks like you are pulling in changes from all over. which makes this almost impossible to review.

**Comment by morgana2313** [comment]
> can you rebase your branch against master from the start? it looks like you are pulling in changes from all over. which makes this almost impossible to review.

Done.

**Comment by morgana2313** [comment]
@whytewolf: I have no idea how to proceed. The pr-opensuse-15-x86_64-py3-pytest says: `Failing after 71m — Run Tests (Fast): error in 'sh' step` but I don't understand how that may be causes by these changes. Can you please advise?

**Comment by morgana2313** [comment]
@whytewolf, @nicholasmhughes: the tests give some error that I think are unrelated to my changes. I don't know how to proceed, please advise.

**Comment by nicholasmhughes** [comment]
re-run pr-windows-2019-x64-py3-pytest

**Comment by morgana2313** [comment]
> re-run pr-windows-2019-x64-py3-pytest

I've triggered a rerun with a new push, nut I don't see how I can trigger a rerun otherwise. I also removed the [WIP] tag in the subject line and made  updates to /docs/man/salt.7 and /changelog

As far as I can see there are no ext_pillar.stack unit tests that I can update for this `__env__` behaviour.

**Comment by whytewolf** [comment]
> > re-run pr-windows-2019-x64-py3-pytest
> 
> I've triggered a rerun with a new push, but I don't see how I can trigger a rerun otherwise. I also removed the [WIP] tag in the subject line and made updates to /docs/man/salt.7 and /changelog
> 
> As far as I can see there are no ext_pillar.stack unit tests that I can update for this `__env__` behaviour.

there may not be an existing test, that does happen. If possible would you be able to make a new test for the functionality?

**Comment by morgana2313** [comment]
> there may not be an existing test, that does happen. If possible would you be able to make a new test for the functionality?

@whytewolf: I've added a stacked pillar unit test, testing the old and new functionality.

**Comment by whytewolf** [comment]
> @whytewolf:
> 
> The friendly PR-robot tells me
> 
> > The function 'ext_pillar' on 'salt/pillar/stack.py' does not have a docstring
> 
> That's true, but the docstring is in the top of the file. Should I move the docstring, add a dummy reference or leave this as it is?
> 
> I've filed bug report #62693 that I discovered working on this PR.
> 
> I consider this PR ready for merging in master.

take inspiration for the other ext_pillar modules. the docstring for the `ext_pillar` function should be short and to the point of what the function does. the one at the top that covers the enter functionality should be left alone. as that is the main point of the entire module. 

the one for ext_pillar should be something like `Execute queries against MySQL, merge and return as a dict`

**Comment by morgana2313** [comment]
@whytewolf: ok I've added the docstring, and all checks pass and I need a review. Can you do the code review?

Regards, Jasper

**Review by whytewolf — DISMISSED** [review]

File: doc/man/salt.7 [hunk 1]
```diff
@@ -353551,15 +353551,16 @@ Here is an example of such a configuration, which should speak by itself:
 .ft C
 ext_pillar:
   \- makostack:
```
Comment (Ch3LL) [inline doc/man/salt.7:353553]: The man pages should not be edited directory. These are auto generated at release time.

**Review by Ch3LL — CHANGES_REQUESTED** [review]
Just a small request

File: doc/man/salt.7 (hunk 1 above)
Comment (morgana2313) [reply doc/man/salt.7]: I rolled back these changes.

**Review by Ch3LL — CHANGES_REQUESTED** [review]
looks like there are still changes to the man pages

**Comment by morgana2313** [comment]
> looks like there are still changes to the man pages

Oh, my bad. Fixed.

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**